translations_parser/parser.py

```
@@ -0,0 +1,228 @@
+                continue
+            text = line[position:]
+
+            # Record logs depending on Marian headers
+            if len(headers) >= 2:
+                # First is task timestamp, second is marian timestamp
+                _, _, *marian_tags = headers
+                tag = "_".join(*marian_tags) if marian_tags else "_default"
+                self.indexed_logs[tag].append(text)
+
+            yield headers, text
+
+    def _parse(self):
+        if self.parsed:
+            raise Exception("The parser already ran.")
+        logs_iter = self._iter_log_entries()
+
+        # Consume first lines until we get the Marian header
+        headers = []
+        while ("marian",) not in headers:
+            headers, text = next(logs_iter)
+
+        # Read Marian runtime logs
+        _, version, version_hash, release_date, *_ = text.split()
+        version = version.rstrip(";")
+        major, minor = map(int, version.lstrip("v").split(".")[:2])
+        if (major, minor) > (1, 10):
+            logger.warning("Parsing logs from Marian > 1.10 may yield inconsistent results")
+
+        # Read Marian execution description on the next lines
+        desc = []
+        for headers, text in logs_iter:
+            if ("marian",) not in headers:
+                break
+            desc.append(text)
+        self.info = {
+            "version": version,
+            "version_hash": version_hash,
+            "release_date": release_date,
+            "description": " ".join(desc),
+        }
+
+        # Try to parse all following config lines as YAML
+        config_yaml = ""
+        while ("config",) in headers:
+            if "Model is being created" in text:
+                headers, text = next(logs_iter)
+                break
+            config_yaml += f"{text}\n"
+            headers, text = next(logs_iter)
+        self.config = yaml.safe_load(config_yaml)
```

You should catch any error here, and raise a meaningful exception ("Invalid `config` section")